### PR TITLE
Solution

### PR DESCRIPTION
**Pull request by dariia-romanova**

### CODE REVIEW

File: src/formatDate.js [hunk 1]
```diff
@@ -50,7 +50,39 @@
...
+  if (+years < 30) {
+    return '20' + years;
+  }
+
+  return '19' + years;
+}
+
+function shortYearsFormat(years) {
+  return years.slice(2);
 }
```
Comment (dmtrhrytsak) [inline src/formatDate.js:86]: You could make one function that would handle all edge cases with years. It's okay this way too, though👍

File: src/formatDate.js [hunk 2]
```diff
@@ -50,7 +50,39 @@
...
+
+  for (let i = 0; i < oldDate.length; i++) {
+    if (fromFormat[i] === 'YY') {
+      dateInfo['YYYY'] = extendYearsFormat(oldDate[i]);
+    } else if (fromFormat[i] === 'YYYY') {
+      dateInfo['YY'] = shortYearsFormat(oldDate[i]);
+    }
+
+    dateInfo[fromFormat[i]] = oldDate[i];
+  }
```
Comment (dmtrhrytsak) [inline src/formatDate.js:67]: It seems like you could iterate using `for of` or `for each` loops)

A rule of thumb: if you don't need any manipulation with indices (terminating earlier, getting the previous/next element [i + 1]/[i - 1] etc.), use either `for of` or `for each` loops or something like that)

**Review by dmtrhrytsak — APPROVED** [review]

File: src/formatDate.js (hunk 2 above)
Comment (dmtrhrytsak) [reply src/formatDate.js:67]: Oops, I overlooked the last statement😅

Comment (AlinaMatuschak) [reply src/formatDate.js:67]: Agree. For of will be better 👌

**Review by AlinaMatuschak — APPROVED** [review]
Awesome 👏

File: src/formatDate.js (hunk 2 above)
Comment (dmtrhrytsak) [reply src/formatDate.js:67]: If ypu need to keep track of the current index, you could try `forEach` it accepts index as the third argument)

**Pull request closed without merging**